packages/frontend/src/components/ProjectAccess/CreateProjectAccessModal.tsx

```diff
@@ -36,60 +39,101 @@ const CreateProjectAccessModal: FC<Props> = ({ projectUuid, onClose }) => {
     const { user } = useApp();
 
     const { data: organizationUsers } = useOrganizationUsers();
-    const { mutateAsync: createMutation, isLoading } =
-        useCreateProjectAccessMutation(projectUuid);
+    const { data: organizationRoles, isLoading: isLoadingRoles } =
+        useOrganizationRoles();
+    const { mutateAsync: upsertRoleMutation, isLoading: isUpsertingRole } =
+        useUpsertProjectUserRoleAssignmentMutation(projectUuid);
+    const { mutateAsync: grantAccessMutation, isLoading: isGrantingAccess } =
+        useGrantProjectAccessWithRoleIdMutation(projectUuid);
 
     const {
         mutateAsync: inviteMutation,
         isLoading: isInvitationLoading,
         reset,
     } = useCreateInviteLinkMutation();
 
-    const form = useForm<Pick<CreateProjectMember, 'email' | 'role'>>({
+    const form = useForm<{ email: string; roleId: string }>({
         initialValues: {
             email: '',
-            role: ProjectMemberRole.VIEWER,
+            roleId: '',
         },
     });
 
     const [addNewMember, setAddNewMember] = useState<boolean>(false);
     const [inviteLink, setInviteLink] = useState<InviteLink | undefined>();
     const [emailOptions, setEmailOptions] = useState<string[]>([]);
 
+    // Prepare role options with grouping
+    const roleOptions = useMemo(() => {
```

Comment:
this logic is already implemented in the project access UI component. we can reuse that one